docs/rules/max-lines-per-function.md

```diff
@@ -0,0 +1,130 @@
+# enforce a maximum function length (max-lines-per-function)
+
+Some people consider large functions a code smell. Large functions tend to do a lot of things and can make it hard following what's going on. While there is not an objective maximum number of lines considered acceptable in a function, most people would agree it should not be above 200. Google's JS style guide (by proxy of their C++ style guide), recommends 40 lines or fewer per function. The Linux Kernel's style guide recommends 48 lines or fewer.
```

Comment:
I'm not quite sold on the description here - it seems too much like a clone of my original feature request (which was written to persuade, not simply explain). This should really be informative, without as much emphasis on other style guides. It's one thing to reference Google's style guide\*, but the kernel style guide reference doesn't belong here.

You might want to look [at this comment](https://github.com/eslint/eslint/issues/9842#issuecomment-379135358) for a few more ideas on what you *could* include. Specifically, it might be worth mentioning how with those, `max-statements` would fail to capture the size of the method in those situations.\*\* (It's worth noting that in that comment, I pointed to only the vdom side, not cases with long method chains, etc.)

\* Preferably with a link to both the JS style guide and the relevant C++ section and with the awkward "by proxy of ..." recast to be a bit easier to follow.

\*\* If you choose to link to those, it'd be better if you linked to the files with exact commit hashes. That way, it doesn't go stale as quickly.

Reply:
I've rewritten the introduction and included an example used from your comment to explain the differences between the other rules. Thanks